packages/components/src/List/Toolbar/ColumnChooserButton/ColumnChooser/ColumnChooser.component.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { ColumnChooserProvider } from './columnChooser.context';
+import { useColumnChooserManager } from '../hooks';
+import FilterBar from '../../../../FilterBar';
+import ColumnChooserBody from './ColumnChooserBody';
+import ColumnChooserFooter from './ColumnChooserFooter';
+import ColumnChooserHeader from './ColumnChooserHeader';
+import getDefaultT from '../../../../translate';
+import cssModule from './ColumnChooser.scss';
+import { getTheme } from '../../../../theme';
+
+const theme = getTheme(cssModule);
+
+const haveColumnLabel = label => column => column.label.toLowerCase().includes(label.toLowerCase());
+const filterColumnsChooser = (columns, filter) => columns.filter(haveColumnLabel(filter));
+const changeVisibleToHidden = column => ({ ...column, hidden: !column.visible });
+const transformColumnsChooser = columns => columns.map(changeVisibleToHidden);
+
+export default function ColumnChooser({
+	children,
+	columns,
+	filterValue,
+	id,
+	nbLockedLeftItems = 0,
+	onClose,
+	submit,
+	t = getDefaultT(),
+}) {
+	const { columnsChooser, onChangeVisibility, onSelectAll, selectAll } = useColumnChooserManager(
+		columns,
+		nbLockedLeftItems,
+	);
+
+	useEffect(() => {
+		// eslint-disable-next-line no-console
+		console.warn(
+			'Guideline and development of the ColumnChooser component still in progress. It may have breaking change in the future',
+		);
+	}, []);
+
+	// We are transforming back the field visible to hidden, to be compliant with the list.
+	const onSubmit = event => {
+		event.preventDefault();
+		submit(event, transformColumnsChooser(columnsChooser));
+	};
+	const [filter, setFilter] = useState(filterValue || '');
+	const onFilter = (_, value) => setFilter(value);
+	const resetFilter = () => setFilter('');
```

Comment:
Can you memoize those 2 functions please

Reply:
Mouerf this is a total no perf gain, the instanciation of this function literaly cost nothing. It's the same old debat about inline function.
Memoize is interesting not for the function instanciation but for the result and avoiding unecessary computing.

Reply:
Do you mean at this scale, re-rendering the filter everytime costs nothing ?
If so you're right, I can close my eyes on this one, but I'll always tell you to optimize :D